### PR TITLE
Cleans up POSIX/C bits in H5private.h

### PR DESCRIPTION
**Pull request by derobins**
* Assume difftime exists (C89)
* Reorg AC_CHECK_HEADERS so headers are in alphabetical order
* Split off networking-related AC_CHECK_HEADERS
* Remove unused UNAME_CYGWIN from configure.ac
* Remove checks for unused sys/timeb.h
* Tidying pass over H5private.h HD prefix macros
* Tidy H5win32defs.h
* Add HD prefix to various scanf calls

### CODE REVIEW

File: src/H5private.h [hunk 1]
```diff
@@ -590,227 +590,231 @@ typedef struct {
...
 #define HDacos(X) acos(X)
-#endif /* HDacos */
+#endif
 #ifndef HDalarm
-#ifdef H5_HAVE_ALARM
 #define HDalarm(N) alarm(N)
-#else /* H5_HAVE_ALARM */
-#define HDalarm(N) (0)
-#endif /* H5_HAVE_ALARM */
-#endif /* HDalarm */
```
Comment (derobins) [inline src/H5private.h:623]: Protected in the one place in the source where it's used.

File: src/H5private.h [hunk 2]
```diff
@@ -590,227 +590,231 @@ typedef struct {
...
 #define HDcuserid(S) cuserid(S)
-#endif /* HDcuserid */
+#endif
 #ifndef HDdifftime
-#ifdef H5_HAVE_DIFFTIME
 #define HDdifftime(X, Y) difftime(X, Y)
-#else /* H5_HAVE_DIFFTIME */
-#define HDdifftime(X, Y) ((double)(X) - (double)(Y))
-#endif /* H5_HAVE_DIFFTIME */
-#endif /* HDdifftime */
```
Comment (derobins) [inline src/H5private.h:732]: difftime is C89

File: src/H5private.h [hunk 3]
```diff
@@ -590,227 +590,231 @@ typedef struct {
...
+#endif
 #ifndef HDfclose
 #define HDfclose(F) fclose(F)
-#endif /* HDfclose */
-#ifdef H5_HAVE_FCNTL
+#endif
 #ifndef HDfcntl
 #define HDfcntl(F, C, ...) fcntl(F, C, __VA_ARGS__)
-#endif /* HDfcntl */
-#endif /* H5_HAVE_FCNTL */
```
Comment (derobins) [inline src/H5private.h:782]: Protected in the source

File: src/H5private.h [hunk 4]
```diff
@@ -824,801 +828,800 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
...
-#ifndef H5_HAVE_WIN32_API
-/* These definitions differ in Windows and are defined in
- * H5win32defs for that platform.
- */
-typedef struct stat h5_stat_t;
-typedef off_t       h5_stat_size_t;
-#define HDoff_t off_t
-#endif /* H5_HAVE_WIN32_API */
-
-#define H5_SIZEOF_H5_STAT_SIZE_T H5_SIZEOF_OFF_T
```
Comment (derobins) [inline src/H5private.h:897]: Moved to top

File: src/H5private.h [hunk 5]
```diff
@@ -824,801 +828,800 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
...
+#endif
 #ifndef HDfstat
 #define HDfstat(F, B) fstat(F, B)
-#endif /* HDfstat */
-#ifndef HDlstat
-#define HDlstat(S, B) lstat(S, B)
-#endif /* HDlstat */
-#ifndef HDstat
-#define HDstat(S, B) stat(S, B)
-#endif /* HDstat */
```
Comment (derobins) [inline src/H5private.h:886]: Put in alphabetical order

File: src/H5private.h [hunk 6]
```diff
@@ -824,801 +828,800 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
...
+#   define HDrandom() rand()
+#   endif
+#   ifndef HDsrand
+#   define HDsrand(S) srand(S)
+#   endif
+#   ifndef HDsrandom
+#   define HDsrandom(S) srand(S)
+#   endif
+#endif
+/* clang-format on */
```
Comment (derobins) [inline src/H5private.h:1275]: Nothing new here - just indenting so it's readable.

File: src/H5private.h [hunk 7]
```diff
@@ -824,801 +828,800 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
...
+#endif
 #ifndef HDstrxfrm
 #define HDstrxfrm(X, Y, Z) strxfrm(X, Y, Z)
-#endif /* HDstrxfrm */
-#ifdef H5_HAVE_SYMLINK
+#endif
 #ifndef HDsymlink
 #define HDsymlink(F1, F2) symlink(F1, F2)
-#endif /* HDsymlink */
-#endif /* H5_HAVE_SYMLINK */
```
Comment (derobins) [inline src/H5private.h:1497]: Protected in the source

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Review by qkoziol — APPROVED** [review]

**Pull request merged**